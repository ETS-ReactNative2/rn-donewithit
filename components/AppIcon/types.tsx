import { StyleProp, TextStyle } from 'react-native'

export type AppIconType = {
  name?: any
  size?: number
  color?: string
  style?: StyleProp<TextStyle>
}
