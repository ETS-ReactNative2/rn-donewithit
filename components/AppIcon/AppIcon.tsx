import React from 'react'
import { MaterialCommunityIcons } from '@expo/vector-icons'
import { AppIconType } from './types'
import CONSTS from './consts'

function AppIcon({
  name = CONSTS.NAME,
  size = CONSTS.SIZE,
  color = CONSTS.COLOR,
  style
}: AppIconType) {
  return (
    <MaterialCommunityIcons
      style={style}
      name={name}
      size={size}
      color={color}
    />
  )
}

export default AppIcon
