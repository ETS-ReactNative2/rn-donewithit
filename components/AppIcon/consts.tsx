import COLORS from '../../config/colors'

const CONSTS = {
  NAME: 'email',
  SIZE: 20,
  COLOR: COLORS.WHITE
}

export default CONSTS
