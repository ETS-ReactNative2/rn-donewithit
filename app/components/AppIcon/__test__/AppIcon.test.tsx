import React from 'react'
import renderer from 'react-test-renderer'
import AppIcon from '../index'
import COLORS from '../../../config/colors'

jest.mock(
  '@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js',
  () => {
    return () => ''
  }
)

const props = {
  name: 'camera',
  size: 20,
  color: COLORS.GRAY
}

describe('AppIcon', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppIcon {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
