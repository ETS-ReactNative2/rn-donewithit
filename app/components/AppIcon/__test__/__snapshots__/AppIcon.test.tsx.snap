// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon renders correctly 1`] = `<Text />`;
